src/platform/EFR32/Efr32PsaOperationalKeystore.cpp

```diff
@@ -0,0 +1,446 @@
+{
+    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Validate public key being activated matches last generated pending keypair
+    VerifyOrReturnError(mPendingKeypair->Pubkey().Matches(nocPublicKey), CHIP_ERROR_INVALID_PUBLIC_KEY);
+
+    mIsPendingKeypairActive = true;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Efr32PsaOperationalKeystore::CommitOpKeypairForFabric(FabricIndex fabricIndex)
+{
+    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(mIsPendingKeypairActive == true, CHIP_ERROR_INCORRECT_STATE);
+
+    // Add key association to key map
+    EFR32OpaqueKeyId id = mPendingKeypair->GetKeyId();
+
+    if (id == kEFR32OpaqueKeyIdUnknown || id == kEFR32OpaqueKeyIdVolatile)
+    {
+        ResetPendingKey();
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    if (mKeyMapping[id - kEFR32OpaqueKeyIdPersistentMin] != kUndefinedFabricIndex)
+    {
+        ResetPendingKey();
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    mKeyMapping[id - kEFR32OpaqueKeyIdPersistentMin] = fabricIndex;
+
+    // Persist key map
+    CHIP_ERROR error = EFR32Config::WriteConfigValueBin(EFR32Config::kConfigKey_OpKeyMap, mKeyMapping, sizeof(mKeyMapping));
+    if (error != CHIP_NO_ERROR)
+    {
+        return error;
+    }
+
+    // There's a good chance we'll need the key again soon
+    mCachedKey->Load(id);
+
+    mPendingKeypair         = nullptr;
+    mIsPendingKeypairActive = false;
+    mPendingFabricIndex     = kUndefinedFabricIndex;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Efr32PsaOperationalKeystore::RemoveOpKeypairForFabric(FabricIndex fabricIndex)
+{
+    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Remove pending keypair if we have it and the fabric ID matches
+    if ((mPendingKeypair != nullptr) && (fabricIndex == mPendingFabricIndex))
+    {
+        RevertPendingKeypair();
+    }
+
+    // Figure out which key ID we're looking for
+    EFR32OpaqueKeyId id = 0;
+    for (size_t i = 0; i < SL_MATTER_MAX_STORED_OP_KEYS; i++)
+    {
+        if (mKeyMapping[i] == fabricIndex)
+        {
+            id = i + kEFR32OpaqueKeyIdPersistentMin;
+
+            // Reset the key mapping since we'll be deleting this key
+            mKeyMapping[i] = kUndefinedFabricIndex;
+            break;
+        }
+    }
+
+    if (id < kEFR32OpaqueKeyIdPersistentMin || id > kEFR32OpaqueKeyIdPersistentMax)
```

Comment:
I think `id > kEFR32OpaqueKeyIdPersistentMax` cannot happen from this loop, and a check of == 0 may be OK as long as `kEFR32OpaqueKeyIdPersistentMin` != 0

Reply:
Made the condition clearer by explicitly using the undefined key ID as marker